profile/meson.build

```diff
@@ -0,0 +1,7 @@
+profile_src = files('profile.c')
+
+libass_profile = executable('profile', profile_src,
+                            install: false,
+                            include_directories: incs,
+                            dependencies: deps,
+                            link_with: libass)
```

Comment:
You have always been able to use `both_libs.get_static_lib()` if you know you have both. If you want to use static libass even when the build is configured to only install shared libs, you need a slightly different tweak.

You can build a `static_library('_libass_objlib', ...., install: false)` and link targets that must build statically to it. And you can convert this into a shared library using extract_all_objects on the internal static library and using it as inputs for a regular `library()` that gets installed.

Reply:
> You have always been able to use both_libs.get_static_lib() if you know you have both.

Does this also work if only-static is built, or must it really be “both”?
If it does this would be great and replicate what libtool does. Basically:
 - if libass is built static-only: use static libass
 - if both variants are built: use static libass *(libtool’s default)*
 - if libass is built shared-only: using shared is ok

Reply:
It has to be either:
- `both_libraries()`
- `library()` with default_library=both

The method only exists for the BothLibrariesHolder internal AST type.

Since you know you use `library()` you can just check the value of default_library, if it is "both" then the method exists. If it is "shared" then there is nothing you can do (and if it did have a method to get a static library it would logically produce an error). If it is "static" then there's nothing you *need* to do.

```meson
if get_option('default_library') == 'both'
    libass_for_tools = libass.get_static_lib()
else
    libass_for_tools = libass
endif
```